File: OC.Domain/Models/Members/MemberTransaction.cs
using OC.Domain.Interfaces;

namespace OC.Domain.Models.Members
{
    public class MemberTransaction : AuditableEntity, IEntityBase
    {
        public long Id { get; set; }
        public long MemberId { get; set; }
        public long TransactionTypeId { get; set; }
        public double Amount { get; set; }
        public string Purpose { get; set; }
        public string Narration { get; set; }
        public DateTime TransactionDateTime { get; set; }

        // Relationships
        public virtual Member? Member { get; set; }
        public virtual TransactionType? TransactionType { get; set; }
    }
}

File: OC.UI/src/app/blci/api/members/member-transaction.ts
import { Member } from "./member";
import { TransactionType } from "./transaction-type";

export interface MemberTransaction {
    id: number;
    memberId: number;
    transactionTypeId: number;
    amount: number;
    purpose: string;
    narration: string;
    transactionDateTime: string;
    member: Member;
    transactionType: TransactionType;
}
